Block/AdvancedCatalog.php

```diff
@@ -195,25 +196,26 @@ protected function getUnusedOption(FilterItem $filterItem)
     /**
      * Get filter items
      *
-     * @return string[]
+     * @return array
      */
     protected function getFilterItems(): array
     {
-        /** @var Navigation $navBlock */
         $navBlock = $this->getNavBlock();
```

Comment:
I suggest to replace everything below this line in this method with a call to `\Shellpea\AdvancedElasticsuiteCatalog\Model\AjaxResponse::getFilterItems()`. Something like this:

```php
    public function __construct(
        \Shellpea\AdvancedElasticsuiteCatalog\Model\AjaxResponseFactory $ajaxResponseFactory
    ) {
        $this->ajaxResponseFactory = $ajaxResponseFactory;
    }

    protected function getFilterItems(): array
    {
        $ajaxResponse = $this->ajaxResponseFactory->create();
        $ajaxResponse->getFilterItems($this->getNavBlock());
    }

```